web-server/src/components/Service/SystemStatus.tsx

```diff
@@ -0,0 +1,167 @@
+import { Box, CircularProgress, Divider } from '@mui/material';
+import { FC, useEffect } from 'react';
+
+import { ServiceNames } from '@/constants/service';
+import { CardRoot } from '@/content/DoraMetrics/DoraCards/sharedComponents';
+import { serviceSlice, ServiceStatusState } from '@/slices/service';
+import { useDispatch, useSelector } from '@/store';
+
+import { FlexBox } from '../FlexBox';
+import { useOverlayPage } from '../OverlayPageContext';
+import { Line } from '../Text';
+
+export const SystemStatus: FC = () => {
+  const dispatch = useDispatch();
+  const loading = useSelector((state) => state.service.loading);
+  const services = useSelector(
+    (state: { service: { services: ServiceStatusState } }) =>
+      state.service.services
+  );
+
+  useEffect(() => {
+    const eventSource = new EventSource(`/api/stream`);
+    eventSource.onmessage = (event) => {
+      const data = JSON.parse(event.data);
+
+      if (data.type === 'status-update') {
+        const statuses = { statuses: data.statuses };
+        dispatch(serviceSlice.actions.setStatus(statuses));
+      }
+
+      if (data.type === 'log-update') {
+        const { serviceName, content } = data;
+        const newLines = content.split('\n');
+        const trimmedLines = newLines.filter(
+          (line: string) => line.trim() !== ''
+        );
+        dispatch(
+          serviceSlice.actions.setServiceLogs({
+            serviceName,
+            serviceLog: trimmedLines
+          })
+        );
+      }
+    };
+
+    eventSource.onerror = () => {
+      eventSource.close();
+    };
+
+    return () => {
+      eventSource.close();
+      dispatch(serviceSlice.actions.resetState());
+    };
+  }, [dispatch]);
+
+  const { addPage } = useOverlayPage();
+
+  const ServiceTitle: { [key: string]: string } = {
+    [ServiceNames.API_SERVER]: 'Backend Server',
+    [ServiceNames.REDIS]: 'Redis Database',
+    [ServiceNames.POSTGRES]: 'Postgres Database',
+    [ServiceNames.SYNC_SERVER]: 'Sync Server'
+  };
+  return (
+    <FlexBox col gap={2} sx={{ padding: '16px' }}>
+      <Line bold white fontSize="24px" sx={{ mb: 2 }}>
+        System Status
+      </Line>
+      <Divider sx={{ mb: 2, backgroundColor: 'rgba(255, 255, 255, 0.2)' }} />
+
+      {loading ? (
+        <Box
+          sx={{
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center'
+          }}
+        >
+          <CircularProgress color="primary" size={50} />
+        </Box>
+      ) : (
+        <FlexBox col gap={2}>
+          {Object.keys(services).map((serviceName) => {
```

Comment:
optional
`{Object.keys(services).map((serviceName: ServiceNames)`

Reply:
got it

Reply:
Hey @e-for-eshaan we cannot directly annotate serviceName as ServiceNames , because Object.keys returns an array of strings,

Reply:
oh, cool then!

Reply:
Great reviews really appreciate it